src/varname.jl

```diff
@@ -414,6 +414,8 @@ end
 
 subsumes_index(i::Colon, ::Colon) = error("Colons cannot be subsumed")
```

Comment:
This line causes downstream issues @yebai

Reply:
It affects every model which uses `:` for indexing.

Reply:
Because we then try to call `subsumes` on it when checking.

Reply:
I might have a fix

Reply:
I remember these issues from the latent Dirichlet allocation model and MCMCChains; IMO, we should always replace `:` with `1:N` where N is a constant.

Reply:
Yeah, but this issue goes a bit deeper than that.

The PR causing these issues were trying to address things like OffsetArrays.jl not working properly, etc. So I think we're now also in need of concretization stuff like `x[2:3]`, etc. I'm trying to figure this out.